internal/controller/sink.go

```diff
@@ -18,50 +20,69 @@ import (
 
 const selectedBackendHeaderKey = "x-envoy-ai-gateway-selected-backend"
 
+// MountedExtProcSecretPath specifies the secret file mounted on the external proc. The idea is to update the mounted
+//
+//	secret with backendSecurityPolicy auth instead of mounting new secret files to the external proc.
+const MountedExtProcSecretPath = "/etc/backend_security_policy" // #nosec G101
+
 // ConfigSinkEvent is the interface for the events that the configSink can handle.
 // It can be either an AIServiceBackend, an AIGatewayRoute, or a deletion event.
 //
 // Exported for internal testing purposes.
 type ConfigSinkEvent any
 
-// configSink centralizes the AIGatewayRoute and AIServiceBackend objects handling
+// ConfigSink centralizes the AIGatewayRoute and AIServiceBackend objects handling
 // which requires to be done in a single goroutine since we need to
 // consolidate the information from both objects to generate the ExtProcConfig
 // and HTTPRoute objects.
-type configSink struct {
-	client client.Client
-	kube   kubernetes.Interface
-	logger logr.Logger
+//
+// Exported for internal testing purposes.
+type ConfigSink struct {
+	client              client.Client
+	kube                kubernetes.Interface
+	logger              logr.Logger
+	defaultExtProcImage string
```

Comment:
Could we also add `defaultExtProcImagePullPolicy corev1.PullPolicy` and value `corev1.PullPolicy(IfNoPresent)`